### PR TITLE
Bug 1304067, added clarifying details to the Secret Restrictions section

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1304067

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@sferich888 PTAL. Thanks!

@adellape @tnguyen-rh @tpoitras @bfallonf Please peer review. Thanks!

**Comment by sferich888** [comment]
LGTM

File: dev_guide/secrets.adoc [hunk 1]
```diff
@@ -119,9 +119,20 @@ Repositories for Builds] for more information.
...
+
+Volume type secrets write data into the container as a file using the volume
+mechanism. *imagePullSecrets* use service accounts for the automatic injection of
+the secret into all pods in a namespaces.
+
+When a template contains a secret definition, the only way for the template to
+use the provided secret is to ensure that the secret volume sources are
+validated to ensure that the specified object reference actually points to an
+object of type `*Secret*`. Therefore, a secret needs to be created before any pods
+that depend on it. Th most effective way to ensure this is to have it
```
Comment (bfallonf) [inline dev_guide/secrets.adoc]: s/th/the

**Comment by bfallonf** [comment]
@ahardin-rh One typo, but LGTM otherwise

File: dev_guide/secrets.adoc (hunk 1 above)
Comment (tnguyen-rh) [inline dev_guide/secrets.adoc]: - s/Th/The/
- Too much "ensure" for me...  If A ensures that EVENT happens, we can try "For EVENT, A must...".

**Comment by tnguyen-rh** [comment]
@ahardin-rh One typo and RFE.  Otherwise LGTM.

**Comment by ahardin-rh** [comment]
[rev_history]
|link:../dev_guide/secrets.html[Secrets]
|Added clarifying details to the link:../dev_guide/secrets.html#secrets-restrictions[Restrictions] section.
%

**Pull request merged**